tracer/src/Datadog.Trace/Configuration/GlobalSettings.cs

```diff
@@ -48,9 +48,14 @@ internal GlobalSettings(IConfigurationSource source)
         public bool DebugEnabled { get; private set; }
 
         /// <summary>
-        /// Gets or sets the global settings instance.
+        /// Gets the configuration source instance.
         /// </summary>
-        internal static GlobalSettings Source { get; set; } = FromDefaultSources();
+        internal static IConfigurationSource ConfigurationSource { get; private set; } = CreateDefaultConfigurationSource();
```

Comment:
nit: given your exposing this, I would probably try to make `CreateDefaultConfigurationSource` private, so that this is the only way to access it. I would probably also move it _off_ `GlobalSettings` to its own static type, as it's not strictly related to this (mostly obsolete) `GlobalSettings` type.

e.g.

```csharp
internal class ConfigurationSource
{
  internal static IConfigurationSource DefaultSources { get; private set; } = CreateDefaultConfigurationSource();
  
  private static CompositeConfigurationSource CreateDefaultConfigurationSource() {}
  
  internal static Reload(){}
}
```

Reply:
I really the idea to separate `ConfigurationSource` from `GlobalSettings`, did it here 45afa07cdc2e0dc1b5c76a1e65b93fd9c34fc07e

However, I think I cannot make `CreateDefaultConfigurationSource` private, since `TracerSettings` use it, and it's public method 😞 

https://github.com/DataDog/dd-trace-dotnet/blob/43d3a240eb2d6fe1c0e7e7941742c901d42441ec/tracer/src/Datadog.Trace/Configuration/TracerSettings.cs#L516:L518

Reply:
But it's not the same functionality 🤔 
Before your change it will enumerate configurations (so if config file changed, it will be represented in returned class)
After you change, it will return once created class.

I don't have strong opinion here, and if you say it's OK, I'll do it.

Reply:
Sure, but it's public, so may be someone using it, and expecting the exact behavior? 😄

Reply:
Oh, and another thing, it won't compile, because of different return type, it expecting `CompositeConfigurationSource`, but `GlobalConfigurationSource.Instance` is `IConfigurationSource`.
We can solve it by using some workarounds, but as discussed, it not worth it, so I'm resolving this conversation.